src/gc-heap-snapshot.cpp

```diff
@@ -206,13 +206,18 @@ size_t record_node_to_gc_snapshot(jl_value_t *a) JL_NOTSAFEPOINT
         self_size = sizeof(jl_svec_t) + sizeof(void*) * jl_svec_len(a);
     }
     else if (jl_is_module(a)) {
-        name = "Module";
+        name = jl_symbol_name_(((_jl_module_t*)a)->name);
         self_size = sizeof(jl_module_t);
     }
     else if (jl_is_task(a)) {
         name = "Task";
         self_size = sizeof(jl_task_t);
     }
+    else if (jl_is_datatype(a)) {
+        auto type = string("Type{") + string(jl_symbol_name_(((_jl_datatype_t*)a)->name->name)) + string("}");
+        name = StringRef(type);
```

Comment:
in C, this explicitly garbage collects the contents of `name` here. You need to move the definition of `type` above the definition of `StringRef name;`

Reply:
The StringRef constructor doesn't create a copy then?

Reply:
it is optimized for the common case, where a copy would be unnecessarily costly (unlike std::string, since is optimized for the opposite, whatever that is)